miden/src/tools/mod.rs

```diff
@@ -0,0 +1,62 @@
+use assembly::Assembler;
+use vm_core::{Operation, ProgramInputs};
+
+#[derive(Debug, PartialEq)]
+pub struct ProgramInfo {
+    total_vm_cycles: usize,
+    total_noops: usize,
+}
+
+impl ProgramInfo {
+    ///Creates a new ProgramInfo object
+    pub fn new(total_vm_cycles: usize, total_noops: usize) -> ProgramInfo {
+        Self {
+            total_vm_cycles,
+            total_noops,
+        }
+    }
+
+    ///Get total vm cycles to execute a program
+    pub fn total_vm_cycles(&self) -> usize {
+        self.total_vm_cycles
+    }
+
+    ///Get total noops executed as part of a program
+    pub fn total_noops(&self) -> usize {
+        self.total_noops
+    }
+}
+
+/// Returns program analysis for a given script string. Returns ProgramInfo with following fields:
+/// - total_vm_cycles (vm cycles it takes to execute the entire script)
+/// - total_noops (total noops executed as part of a program)
+pub fn analyze(program: String) -> ProgramInfo {
```

Comment:
I think this should probably take `ProgramInputs` as a parameter for programs which cannot be executed against blank inputs.

Reply:
For the binary, do we need to allow `advice_tape` and `advice_sets` as input as well? Currently it only takes optional `stack_init` as external input from user.

Reply:
I mentioned this in a comment below, but let's not worry about this for now.